### PR TITLE
MAID-1847 Add a Section authority

### PR DESCRIPTION
**Pull request by fizyk20**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/routing_table/mod.rs [hunk 1]
```diff
@@ -396,10 +363,8 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         self.closest_names(name, count).is_some()
     }
 
-    /// Returns the `count` closest entries to `name` in the routing table, including our own name,
-    /// sorted by ascending distance to `name`. If we are not close, returns `None`.
-    pub fn closest_names(&self, name: &T, count: usize) -> Option<Vec<&T>> {
-        let result = self.groups
+    fn all_closest_names(&self, name: &T, count: usize) -> Vec<&T> {
```
Comment (afck) [inline src/routing_table/mod.rs]: Maybe `closest_known_names`, as opposed to the real `closest_names`, where we know that they're the closest in the whole network?

File: src/routing_table/mod.rs [hunk 2]
```diff
@@ -753,46 +724,53 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
...
                    exclude: T,
                    route: usize)
                    -> Result<HashSet<T>, Error> {
-        let (closest_group, target_name) = match *dst {
-            Destination::Group(ref target_name) => {
-                let (prefix, group) = self.closest_group(target_name);
-                if *prefix == self.our_group_prefix {
-                    return Ok(group.clone());
+        let target_name = dst.name();
+        let (closest_section, target_name) = if dst.is_section() || dst.is_group() {
```
Comment (afck) [inline src/routing_table/mod.rs]: We set `target_name` to `target_name` on both brances now, I think, so it doesn't need to be assigned here.

File: src/routing_table/mod.rs (hunk 1 above)
Comment (fizyk20) [inline src/routing_table/mod.rs]: Good idea.

File: src/routing_table/authority.rs [hunk 3]
```diff
@@ -46,22 +47,47 @@ pub enum Authority {
...
+            Authority::ClientManager(_) |
+            Authority::NaeManager(_) |
+            Authority::NodeManager(_) |
+            Authority::ManagedNode(_) |
+            Authority::Client { .. } => false,
+            Authority::Section(_) => true,
+        }
+    }
+
+    /// Returns `true` if the destination is an individual node, and `false` otherwise.
```
Comment (fizyk20) [inline src/routing_table/authority.rs]: I'm fine with either, with a slight preference for putting the backticks (although, TBH, I haven't been paying much attention to it so far).

File: src/routing_table/mod.rs (hunk 2 above)
Comment (fizyk20) [inline src/routing_table/mod.rs]: Well spotted.

**Pull request merged**